plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcRecordSet.java

```diff
@@ -52,13 +57,15 @@ public void setUp()
         table = database.getTableHandle(SESSION, new SchemaTableName("example", "numbers"));
         split = database.getSplit(SESSION, table);
         columnHandles = database.getColumnHandles(SESSION, table);
+        executor = newCachedThreadPool(daemonThreadsNamed("TestJdbcRecordSet-%s"));
     }
 
     @AfterClass(alwaysRun = true)
     public void tearDown()
             throws Exception
     {
         database.close();
+        assertTrue(executor.shutdownNow().isEmpty());
```

Comment:
just `executor.shutdownNow()` -- it's tests' responsibility to wait if they need to

also, shutdown also if database.close() fails -- `io.airlift.testing.Closeables#closeAll`

Reply:
> just executor.shutdownNow() -- it's tests' responsibility to wait if they need to

I know. I wanted to double check if test are not written incorrectly.

Reply:
in the case where this actually could trigger, we would most likely end up with flaky test, and we wouldn't even know which one

please do not do this